worldedit-bukkit/src/main/java/com/sk89q/worldedit/bukkit/BukkitCommandSender.java

```diff
@@ -42,15 +42,15 @@
      */
     private static final UUID DEFAULT_ID = UUID.fromString("a233eb4b-4cab-42cd-9fd9-7e7b9a3f74be");
 
+    private final Audience audience;
     private final CommandSender sender;
-    private final WorldEditPlugin plugin;
 
     public BukkitCommandSender(WorldEditPlugin plugin, CommandSender sender) {
         checkNotNull(plugin);
         checkNotNull(sender);
         checkArgument(!(sender instanceof Player), "Cannot wrap a player");
 
-        this.plugin = plugin;
+        this.audience = BukkitAudiences.create(plugin).audience(sender);
```

Comment:
You should have one instance of `BukkitAudiences.create(plugin)` cached and re-use the created factory to create your audiences.